### PR TITLE
Correct scattering index and enable error inflation in ATMS data,

### PR DESCRIPTION
**Pull request by jianjunj**
**Description**

The scattering index was not correctly calculated by using a wrong channel at 89GHz.  The error inflation referring to it was turned off instead of correcting the calculation.  This PR corrects the scattering index and enables error inflation in ATMS observations.

Note, no test has been conducted. However, it very likely creates non-zero difference in GSI results.

**Type of change**

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Checklist**

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by ADCollard — APPROVED** [review]
An important bug fix.  Thanks.

**Review by emilyhcliu — APPROVED** [review]
Good Catch!

**Comment by RussTreadon-NOAA** [comment]
@jianjunj , please follow the GSI code management procedures on the [_GSI:-How-to-Make-Chhttps://github.com/NOAA-EMC/GSI/wiki/GSI:-How-to-Make-Changesanges_](url) for future GSI PRs. Changes to the GSI begin with opening an issue.  Test results, both ctests and other necessary tests to validate the correctness of the change, are recorded in the issue (or PR).

You don't need to create an issue for this PR but please do so for future GSI changes. We do, however, need to see ctest and other validation results in this PR. ctests need to be run at a minimum on WCOSS2 and Hera.  Running ctests on WCOSS2, Hera, Hercules, and Orion is preferable.

File: src/gsi/clw_mod.f90 [hunk 1]
```diff
@@ -1961,6 +1961,7 @@ subroutine ret_amsua(tb_obs,nchanl,tsavg5,zasat,clwp_amsua,ierrret,scat)
 ! Declare local variables 
   real(r_kind) :: d0, d1, d2, coszat
 ! real(r_kind) :: c0, c1, c2
+  real(r_kind) :: tb890 = 0.0
```
Comment (RussTreadon-NOAA) [inline src/gsi/clw_mod.f90]: Replace `0.0` with `zero`

File: src/gsi/clw_mod.f90 [hunk 2]
```diff
@@ -1979,8 +1980,17 @@ subroutine ret_amsua(tb_obs,nchanl,tsavg5,zasat,clwp_amsua,ierrret,scat)
...
-      scat=-113.2_r_kind+(2.41_r_kind-0.0049_r_kind*tb_obs(1))*tb_obs(1)  &
-           +0.454_r_kind*tb_obs(2)-tb_obs(15)
+      if (nchanl == 15) then
+!         AMSU-A
+          tb890 = tb_obs(15)
+      else if (nchanl == 22) then
+!         ATMS
+          tb890 = tb_obs(16)
+      endif
+      if (tb890 > 0.0) then
```
Comment (RussTreadon-NOAA) [inline src/gsi/clw_mod.f90]: Replace `0.0` with `zero`

**Review by RussTreadon-NOAA — COMMENTED** [review]
Minor changes to adhere to [GSI code standards](https://github.com/NOAA-EMC/GSI/wiki/GSI---Code-Standards).

**Comment by RussTreadon-NOAA** [comment]
**WCOSS2 ctests**

Install `jianjunj:feature/atms_sacttering_index` at fb65b29a as _updat_ and `develop` at 9ada88e8 as _contrl_ on Dogwood.  Run ctests with the following results
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/pr812/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  1689.87 sec
2/6 Test #6: global_enkf ......................   Passed  1753.82 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  2116.57 sec
4/6 Test #4: hafs_4denvar_glbens ..............   Passed  2237.49 sec
5/6 Test #2: rtma .............................   Passed  2403.79 sec
6/6 Test #1: global_4denvar ...................***Failed  4926.59 sec

83% tests passed, 1 tests failed out of 6

Total Test time (real) = 4926.82 sec
```
The `global_4denvar` failure is due to
```
The results (penalty) between the two runs are nonreproducible,
thus the regression test has Failed on cost for global_4denvar_loproc_updat and global_4denvar_loproc_contrl analyses.
```
This failure is expected.  This PR corrects a bug in the ATMS scattering index and error inflation.  Comparison of the _contrl_ and _updat_ `fort.207` shows initial differences in the ATMS penalties.  The _contrl_ run has
```
o-g 01 rad   npp          atms           15399934         3454         2590    675.79       675.79      0.26092      0.26092    
o-g 01 rad   n20          atms           15399934         3432         2594    632.69       632.69      0.24391      0.24391    
o-g 01 rad   n21          atms           15399934         3432         2619    628.07       628.07      0.23981      0.23981    
```
whereas the _updat_ run has
```
o-g 01 rad   npp          atms           15399934         3454         2590    673.24       673.24      0.25994      0.25994    
o-g 01 rad   n20          atms           15399934         3432         2594    618.16       618.16      0.23830      0.23830    
o-g 01 rad   n21          atms           15399934         3432         2619    621.66       621.66      0.23736      0.23736  
```
The number of assimilated ATMS observations is identical between the two `gsi.x` executables.  The _updat_ `gsi.x` has slightly smaller penalties.  This reflects corrected error inflation and use of the correct scattering index.

Interestingly, the `hafs_3denvar_hybens` and `rrfs_3denvar_rdasens` tests also assimilate ATMS radiances but analysis results are identical between _contrl_ and _updat_ `gsi.x`.  Presumably this is because regional DA does not exercise the code altered by this PR.  Is this correct @jianjunj?

**Comment by jianjunj** [comment]
@RussTreadon-NOAA Thank you so much for running the ctests.  I'll do these test in future PRs. In regards to the zero-differences in hafs_3denvar_hybens and rrfs_3denvar_rdasens tests, I think, it is because that this bug fix is not triggered in this test.  It needs the "icw" column to be "1" in "satinfo.txt" to set lcloud_fwd=true and trigger this error inflation over "water" surface: 
https://github.com/NOAA-EMC/GSI/pull/812/files#diff-fb4541d749f4c3618c014384b495bcb3c96136d35b21c049fd6c68d08ca47d1cR3664

**Comment by RussTreadon-NOAA** [comment]
Thank you, @jianjunj, for confirming that the **_no difference_** results for _hafs_3denvar_hybens_ and _rrfs_3denvar_rdasens_ are expected and are correct.

File: src/gsi/clw_mod.f90 (hunk 2 above)
Comment (jianjunj) [reply src/gsi/clw_mod.f90]: Done.

File: src/gsi/clw_mod.f90 (hunk 1 above)
Comment (jianjunj) [reply src/gsi/clw_mod.f90]: Done.

**Review by RussTreadon-NOAA — APPROVED** [review]
Update working copy of `jianjunj:feature/atms_sacttering_index` to 
101e471.  Recompile and rerun `global_4denvar` ctest.  Test still failed (expected).  Analysis results from the previous _updat_ and 101e471 _updat_ are identical.  This is expected, but it's good to have confirmation.

Approve.

**Comment by RussTreadon-NOAA** [comment]
@jianjunj , this PR can not move forward until ctests are run on at least Hera and preferably Hera and one of the MSU machines (Orion or Hercules).

**Comment by jianjunj** [comment]
@RussTreadon-NOAA  I totally understand.

**Comment by RussTreadon-NOAA** [comment]
Install `develop` at [`6fd4bd5`](https://github.com/NOAA-EMC/GSI/commit/6fd4bd568d69f2e80c62f8db205fae2aadf5d506) and `jianjunj:feature/atms_sacttering_index` at [`03f4e72`](https://github.com/jianjunj/GSI_JianjunJin/commit/03f4e72d0f4a7dd3054cd4278448f78db6a2b777) on WCOSS2 (Dogwood), Hera, and Orion.  Ctests launched on each machine.  Results will be posted as ctests complete.

**Comment by RussTreadon-NOAA** [comment]
**WCOSS2 (Dogwood) ctests**

Ctest results on Dogwood are as follows
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/pr812/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  847.98 sec
2/6 Test #6: global_enkf ......................   Passed  1032.32 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  1277.50 sec
4/6 Test #2: rtma .............................   Passed  1330.28 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  1461.74 sec
6/6 Test #1: global_4denvar ...................***Failed  1985.98 sec

83% tests passed, 1 tests failed out of 6

Total Test time (real) = 1985.99 sec

The following tests FAILED:
          1 - global_4denvar (Failed)
```
The `global_4denvar` test failed due to
```
The results (penalty) between the two runs are nonreproducible,
thus the regression test has Failed on cost for global_4denvar_loproc_updat and global_4denvar_loproc_contrl analyses.
```
Examination of the _contrl_ (`<`) and _updat_ (`>`) output show differences in the initial processing of ATMS observations
```
< o-g 01 rad   npp          atms           15399934         3454         2590    675.79       675.79      0.26092      0.26092    
< o-g 01 rad   n20          atms           15399934         3432         2594    632.69       632.69      0.24391      0.24391    
> o-g 01 rad   npp          atms           15399934         3454         2590    673.24       673.24      0.25994      0.25994    
> o-g 01 rad   n20          atms           15399934         3432         2594    618.16       618.16      0.23830      0.23830    
< o-g 01 rad   n21          atms           15399934         3432         2619    628.07       628.07      0.23981      0.23981    
> o-g 01 rad   n21          atms           15399934         3432         2619    621.66       621.66      0.23736      0.23736    
```
The amount of assimilated ATMS data between _contrl_ and _updat_ is the same but the penalties differ with _updat_ penalties being smaller.   This likely reflects ATMS updates in this PR.  Is this a valid statement, @jianjunj?

**Comment by RussTreadon-NOAA** [comment]
**Orion ctests**

Ctest results on Orion are as follows

```
Test project /work2/noaa/da/rtreadon/git/gsi/pr812/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #6: global_enkf ......................   Passed  787.83 sec
2/6 Test #3: rrfs_3denvar_rdasens .............   Passed  969.78 sec
3/6 Test #2: rtma .............................   Passed  1687.74 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  2903.80 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  3022.48 sec
6/6 Test #1: global_4denvar ...................***Failed  3602.80 sec

83% tests passed, 1 tests failed out of 6

Total Test time (real) = 3602.84 sec

The following tests FAILED:
          1 - global_4denvar (Failed)
```
The `global_4denvar` test failed due to 
```
The results (penalty) between the two runs are nonreproducible,
thus the regression test has Failed on cost for global_4denvar_loproc_updat and global_4denvar_loproc_contrl analyses.
```
This is the same reason for the WCOSS2 `global_4denvar` failure.  A check of the _contrl_ and _updat_ `stdout` shows that analysis differences originate with ATMS observations.  Below is the difference between _contrl_ (`<`) and _updat_ (`>`) `fort.207 statistics for the initial innovations
```
< o-g 01 rad   npp          atms           15399934         3454         2590    675.79       675.79      0.26092      0.26092
< o-g 01 rad   n20          atms           15399934         3432         2594    632.69       632.69      0.24391      0.24391
> o-g 01 rad   npp          atms           15399934         3454         2590    673.24       673.24      0.25994      0.25994
> o-g 01 rad   n20          atms           15399934         3432         2594    618.16       618.16      0.23830      0.23830
< o-g 01 rad   n21          atms           15399934         3432         2619    628.07       628.07      0.23981      0.23981
> o-g 01 rad   n21          atms           15399934         3432         2619    621.66       621.66      0.23736      0.23736
```
This behavior is identical to what is observed on WCOSS2.

**Comment by RussTreadon-NOAA** [comment]
**Hera ctests**

Ctest results on Hera are as follows

```
Test project /scratch1/NCEPDEV/da/Russ.Treadon/git/gsi/pr812/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  1332.60 sec
2/6 Test #6: global_enkf ......................   Passed  3751.26 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  4107.00 sec
4/6 Test #2: rtma .............................   Passed  4752.62 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  5311.49 sec
6/6 Test #1: global_4denvar ...................***Failed  5406.08 sec

83% tests passed, 1 tests failed out of 6

Total Test time (real) = 5406.09 sec

The following tests FAILED:
          1 - global_4denvar (Failed)
```
The `global_4denvar` test failed due to
```
The results (penalty) between the two runs are nonreproducible,
thus the regression test has Failed on cost for global_4denvar_loproc_updat and global_4denvar_loproc_contrl analyses.
```
The `fort.207` differences between _contrl_ (`<`) and _updat_ (`>`) are the same as what is observed on WCOSS2 and Orion
```
< o-g 01 rad   npp          atms           15399934         3454         2590    675.79       675.79      0.26092      0.26092
< o-g 01 rad   n20          atms           15399934         3432         2594    632.69       632.69      0.24391      0.24391
> o-g 01 rad   npp          atms           15399934         3454         2590    673.24       673.24      0.25994      0.25994
> o-g 01 rad   n20          atms           15399934         3432         2594    618.16       618.16      0.23830      0.23830
< o-g 01 rad   n21          atms           15399934         3432         2619    628.07       628.07      0.23981      0.23981
> o-g 01 rad   n21          atms           15399934         3432         2619    621.66       621.66      0.23736      0.23736
```

**Comment by RussTreadon-NOAA** [comment]
@jianjunj , the following has been done
1. `jianjunj:feature/atms_sacttering_index` updated with recent changes from `develop`.  Done at [`03f4e72`](https://github.com/NOAA-EMC/GSI/pull/812/commits/03f4e72d0f4a7dd3054cd4278448f78db6a2b777)
2. `jianjunj:feature/atms_sacttering_index` installed on WCOSS2 (Dogwood), Hera, and Orion
3. ctests run on each machine and results reported in this PR.  

If you can confirm that the observed ctest behavior is acceptable, we can move this PR forward.

**Comment by jianjunj** [comment]
@RussTreadon-NOAA Thank you very much fort doing the tests!  Yes, it is expected to have slightly larger ATMS observational error due to the inflation. @ADCollard @emilyhcliu

**Pull request merged**